React/Base/RCTJavaScriptLoader.mm

```diff
@@ -142,10 +142,19 @@ + (NSData *)attemptSynchronousLoadOfBundleAtURL:(NSURL *)scriptURL
     return nil;
 
   case facebook::react::ScriptTag::BCBundle:
-    if (header.BCVersion != runtimeBCVersion) {
+    if (runtimeBCVersion == JSNoBytecodeFileFormatVersion || runtimeBCVersion < 0) {
+      if (error) {
+        *error = [NSError errorWithDomain:RCTJavaScriptLoaderErrorDomain
+                                     code:RCTJavaScriptLoaderErrorBCNotSupported
+                                 userInfo:@{NSLocalizedDescriptionKey:
+                                              @"Bytecode bundles are not supported by this runtime."}];
+      }
+      return nil;
+    }
+    else if ((uint32_t)runtimeBCVersion != header.BCVersion) {
       if (error) {
         NSString *errDesc =
-          [NSString stringWithFormat:@"BC Version Mismatch. Expect: %d, Actual: %d",
+          [NSString stringWithFormat:@"BC Version Mismatch. Expect: %d, Actual: %u",
                     runtimeBCVersion, header.BCVersion];
```

Comment:
Maybe - to be honest I'm not a fan of magic values that sit inside the theoretical range of versions. Personally I'd use a separate argument or box it in an `NSNumber *` (with `nil` indicating incompatibility), but I was trying to keep this PR minimal and minimally opinionated ;).

Looking at it, changing versions to uints across the board would have implications for `RSCJSCExecutor`, `ReactCommon` and a few public APIs - so perhaps the safest thing to do is to keep a light touch?

Your call though.